seq2science/rules/configuration_generic.smk

```diff
@@ -358,6 +365,67 @@ for sample in samples.index:
                          f"access is currently not supported. We advise you to download the sample "
                          f"manually, and continue the pipeline from there on.")
 
+trace_tp.close()
+eutils_tp.close()
+
+def url_is_alive(url):
```

Comment:
hope you don't mind some unasked advise:

I've made a similar function in genomepy, and noticed certain websites get hung up with too many requests. This function seemed to be fairly robust for me:
```
def check_url(url, time_out=3, max_tries=1, _try=1):
    """Check if URL works. Returns bool"""
    try:
        ret = urllib.request.urlopen(url, timeout=time_out)
        # check return code for http(s) urls
        if url.startswith("ftp") or ret.getcode() == 200:
            return True

    except (urllib.request.URLError, socket.timeout):
        # Some providers get swamped with requests, so we allow retries
        if _try <= max_tries:
            time.sleep(1)
            return check_url(url, time_out=time_out, max_tries=max_tries, _try=_try + 1)
        return False
```

Reply:
Yes, I was planning to implement something like this.

Why did you make this recursive and not just sequential?

Reply:
so you can set the maximum number of tries as a variable. The code stays clean even if you want to give a URL 10 tries.

Reply:
Wouldn't this work?
```
for try in range(max_tries):
    try:
        ret = urllib.request.urlopen(url, timeout=time_out)
        # check return code for http(s) urls
        if url.startswith("ftp") or ret.getcode() == 200:
            return True
    except (urllib.request.URLError, socket.timeout):
        continue
    return False
```

Reply:
Not saying either is better, but I usually struggle with implementing recursive stuff :P

Reply:
yes it would. this just popped in my head faster

don't forget to add a pause in the except though!